streamlocal.go

```diff
@@ -227,3 +209,31 @@ func unlink(path string) error {
 		}
 	}
 }
+
+func DefaultSocketListener(socketPath string) (net.Listener, error) {
+	// Create socket parent dir if not exists.
+	parentDir := filepath.Dir(socketPath)
+	err := os.MkdirAll(parentDir, 0700)
+	if err != nil {
+		// TODO: log mkdir failure
+		return nil, nil
```

Comment:
These should return errors and the log todo should be removed here. The log todo can stay in the actual HandleSSHRequest code though